### PR TITLE
[9.0][ADD] stock_available_global: New module

### PR DESCRIPTION
**Pull request by carlosdauden**
This module adds new fields to show product stock in all companies.
In product forms appear two new smarts buttons *On Hand* and *Forecast* witch computes all companies stock.

@Tecnativa

### CODE REVIEW

File: stock_available_global/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+======================
+Stock available global
+======================
+
+This module adds new fields to show product stock in all companies.
+
+In product forms appear two new smarts buttons *On Hand* and *Forecast* witch
```
Comment (pedrobaeza) [inline stock_available_global/README.rst]: s/witch/which

File: stock_available_global/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        search='_search_virtual_available_global',
+        digits=dp.get_precision('Forecasted Quantity'),
+        string='Global Forecasted Quantity',
+        help="Compute all companies")
+
+    @api.multi
+    @api.depends('qty_available', 'virtual_available', 'incoming_qty',
+                 'outgoing_qty')
+    def _product_available_global(self):
+        for product in self:
```
Comment (pedrobaeza) [inline stock_available_global/models/product_product.py]: Better to do:
```
for product in self.sudo():
```
and you save the rest of the sudos.

File: stock_available_global/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        search='_search_virtual_available_global',
+        digits=dp.get_precision('Forecasted Quantity'),
+        string='Global Forecasted Quantity',
+        help="Compute all companies")
+
+    @api.multi
+    @api.depends('qty_available', 'virtual_available', 'incoming_qty',
+                 'outgoing_qty')
+    def _product_available_global(self):
+        for product in self:
```
Comment (pedrobaeza) [inline stock_available_global/models/product_template.py]: `for product in self.sudo():`

File: stock_available_global/views/product_product_view.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record model="ir.ui.view" id="product_normal_form_view">
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="stock.product_form_view_procurement_button" />
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='%(stock.action_stock_level_forecast_report_product)d']"
```
Comment (pedrobaeza) [inline stock_available_global/views/product_product_view.xml:10]: Is this showing all the quants?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: stock_available_global/views/product_product_view.xml (hunk 4 above)
Comment (carlosdauden) [reply stock_available_global/views/product_product_view.xml:10]: No, only shows actual company quants.

Comment (pedrobaeza) [reply stock_available_global/views/product_product_view.xml:10]: Then I think it's better not to offer the clickable possibility and only show data.

**Comment by pedrobaeza** [comment]
@carlosdauden please make the requested changes.

**Comment by pedrobaeza** [comment]
Closing as target branch has changed

**Pull request closed without merging**